repl/repl.go

```diff
@@ -0,0 +1,287 @@
+type repl struct {
+	commands []command
+	client   Client
+	input    string
+}
+
+// Client interface to REPL clients.
+type Client interface {
+	CreateAccount(generatePassphrase bool, accountInfo string) error
+	LocalAccount() *accounts.Account
+	Unlock(passphrase string) error
+	IsAccountUnLock(id string) bool
+	Lock(passphrase string) error
+	AccountInfo(id string)
+	Transfer(from, to, amount, passphrase string) error
+	SetVariables(params, flags []string) error
+	GetVariables() map[string]string
+	Restart(params, flags []string) error
+	NeedRestartNode(params, flags []string) bool
+	Setup(allocation string) error
+}
+
+// Start starts REPL.
+func Start(c Client) {
+	if !TestMode {
+		err := setVariables(c.GetVariables())
+		if err != nil {
+			log.Debug(err.Error())
+		}
+
+		r := &repl{client: c}
+		r.initializeCommands()
+
+		runPrompt(r.executor, r.completer, r.firstTime, uint16(len(r.commands)))
+	} else {
+		// holds for unit test purposes
+		hold := make(chan bool)
+		<-hold
+	}
+}
+
+func (r *repl) initializeCommands() {
+	r.commands = []command{
+		{"create account", "Create account.", r.createAccount},
+		{"unlock account", "Unlock account.", r.unlockAccount},
+		{"lock account", "Lock Account.", r.lockAccount},
+		{"account", "Shows basic local account info about the local account.", r.account},
+		{"transfer coins", "Transfer coins between 2 accounts.", r.transferCoins},
+		{"setup", "Setup POST.", r.setup},
+		{"restart node", "Restart node.", r.restartNode},
+		{"set", "change CLI flag or param. E.g. set param a=5 flag c=5 or E.g. set param a=5", r.setCLIFlagOrParam},
+		{"echo", "Echo runtime variable.", r.echoVariable},
+	}
+}
+
+func (r *repl) executor(text string) {
+	for _, c := range r.commands {
+		if len(text) >= len(c.text) && text[:len(c.text)] == c.text {
+			r.input = text
+			log.Debug(userExecutingCommandMsg, c.text)
+			c.fn()
+			return
+		}
+	}
+
+	fmt.Println(printPrefix, "invalid command.")
+}
+
+func (r *repl) completer(in prompt.Document) []prompt.Suggest {
+	suggets := make([]prompt.Suggest, 0)
+	for _, command := range r.commands {
+		s := prompt.Suggest{
+			Text:        command.text,
+			Description: command.description,
+		}
+
+		suggets = append(suggets, s)
+	}
+
+	return prompt.FilterHasPrefix(suggets, in.GetWordBeforeCursor(), true)
+}
+
+func (r *repl) firstTime() {
+	createNewAccount := yesOrNoQuestion(welcomeMsg) == "y"
+	if createNewAccount {
+		r.createAccount()
+	}
+}
+
+func (r *repl) createAccount() {
+	generatePassphrase := yesOrNoQuestion(generateMsg) == "y"
+	accountInfo := prompt.Input(prefix+accountInfoMsg,
+		emptyComplete,
+		prompt.OptionPrefixTextColor(prompt.LightGray))
+
+	err := r.client.CreateAccount(generatePassphrase, accountInfo)
+	if err != nil {
+		log.Debug(err.Error())
+	} else {
+		r.setup()
+	}
+}
+
+func (r *repl) unlockAccount() {
+	passphrase := r.commandLineParams(1, r.input)
+	err := r.client.Unlock(passphrase)
+	if err != nil {
+		log.Debug(err.Error())
+	} else {
```

Comment:
Again , in idomatic go 
```
if err != nil { 
  log.Debug(err)
  return
}
// The code that was on the else
```

*Note*: this command refers to rest of the places you do the same.

Please check https://golang.org/doc/effective_go.html#errors